js/ui/number_box/number_box.mask.js

```diff
@@ -72,12 +74,13 @@ var NumberBoxMask = NumberBoxBase.inherit({
             return;
         }
 
-        if(browser.msie) {
-            clearTimeout(this._ieCaretTimeout);
-            this._ieCaretTimeout = setTimeout(this._moveCaretToBoundary.bind(this, MOVE_BACKWARD, e));
-        } else {
-            this._moveCaretToBoundary(MOVE_BACKWARD, e);
-        }
+        var that = this;
+
+        this.clearCaretTimeout();
+        this._caretTimeout = setTimeout(function() {
```

Comment:
Note about tests: you should update clock after using KeyboardMock methods like `type`, `press` etc.